include/jemalloc/internal/extent_inlines.h

```diff
@@ -49,32 +49,20 @@ int extent_snad_comp(const extent_t *a, const extent_t *b);
 #if (defined(JEMALLOC_ENABLE_INLINE) || defined(JEMALLOC_EXTENT_C_))
 JEMALLOC_INLINE arena_t *
 extent_arena_get(const extent_t *extent) {
-	return extent->e_arena;
-}
-
-JEMALLOC_INLINE void *
-extent_base_get(const extent_t *extent) {
-	assert(extent->e_addr == PAGE_ADDR2BASE(extent->e_addr) ||
-	    !extent->e_slab);
-	return PAGE_ADDR2BASE(extent->e_addr);
-}
-
-JEMALLOC_INLINE void *
-extent_addr_get(const extent_t *extent) {
-	assert(extent->e_addr == PAGE_ADDR2BASE(extent->e_addr) ||
-	    !extent->e_slab);
-	return extent->e_addr;
-}
-
-JEMALLOC_INLINE size_t
-extent_size_get(const extent_t *extent) {
-	return extent->e_size;
+	unsigned arena_ind = (unsigned)((extent->e_bits &
+	    EXTENT_BITS_ARENA_MASK) >> EXTENT_BITS_ARENA_SHIFT);
+	if (arena_ind > MALLOCX_ARENA_MAX) {
```

Comment:
We convert ```NULL``` to all 1 bits in the setter, so if anyone reads the value back out, we need to convert it back to ```NULL```.  I didn't check whether we ever actually call this getter when the field is encoding ```NULL```.

Re: the ```cassert()``` suggestion, the ```EXTENT_BITS_*``` definitions are based on ```MALLOCX_ARENA_BITS```, so the defined relationship with ```MALLOCX_ARENA_MAX``` assures we have enough bits to store the arena index.